### PR TITLE
Allow MultiNumberRange to be inclusive or exclusive

### PR DESCRIPTION
**Pull request by jmancewicz**
The Learning Rate range validator has a min of 0.0 inclusive, but a learning rate of 0 seems bad, (I may be wrong), so, I've added min_inclusive and max_inclusive booleans to that 0 is excluded.

### CODE REVIEW

**Comment by lukeyeager** [comment]
Actually I've found it useful to have a learning rate of zero before. I was importing a pre-trained model and didn't want to change the weights at all (granted, we should just create an interface for doing that).

**Comment by gheinrich** [comment]
Maybe this shouldn't apply to the learning rate as @lukeyeager has an ingenious use case there but exclusive fields sound like a useful thing to have. I think it's a good idea.

File: digits/utils/forms.py [hunk 1]
```diff
@@ -339,26 +339,39 @@ class MultiNumberRange(object):
...
-            if data is None or (self.min is not None and data < self.min) or \
-                    (self.max is not None and data > self.max):
+            print 'checking', data, self.min
+            flags = 0
+            flags |= (data is None) << 0
+            flags |= (self.min is not None and self.min_inclusive and data < self.min) << 1
+            flags |= (self.max is not None and self.max_inclusive and data > self.max) << 2
+            flags |= (self.min is not None and not self.min_inclusive and data <= self.min) << 3
+            flags |= (self.max is not None and not self.max_inclusive and data >= self.max) << 4
+
```
Comment (gheinrich) [inline digits/utils/forms.py]: Wondering why you are test the condition here and not directly below (i.e. without setting `flags` as an intermediate step)? I don't mind either way, I am just wondering if there is a trick to learn there :-)

Comment (jmancewicz) [inline digits/utils/forms.py]: I had borrowed if from the wtforms case where there is only a min and max, and it has logic for cases min!=None, max!=None, and (min!=None && max!=None).  I had initially started working on a message such that there was one message that handled any permutation of settings, that's when I added the flags, and decided against it.  I don't find the flags offensive, but would be happy to change them if it's confusing or messy.

**Comment by jmancewicz** [comment]
@lukeyeager, I just KNEW there would be a case for learning rate == 0!

So, I'll remove it from the learning rate field, but I'll keep it around and maybe make number range valuators with this option in the future.

**Comment by lukeyeager** [comment]
> So, I'll remove it from the learning rate field

ping

**Comment by jmancewicz** [comment]
@lukeyeager, I removed this from the learning rate.

**Comment by lukeyeager** [comment]
> I removed this from the learning rate.

Thanks!
